src/Compilers/Core/Portable/Compilation/SemanticModel.cs

```diff
@@ -367,6 +367,23 @@ public SemanticModel ParentModel
             get;
         }
 
+        /// <summary>
+        /// If this is a member semantic model for executable code, then returns the syntax tree semantic model that was used to create this model.
+        /// If this is a syntax tree semantic model, then returns this instance.
+        /// Otherwise, returns null.
+        /// </summary>
+        internal SemanticModel OriginalSyntaxTreeModel => OriginalSyntaxTreeModelCore;
+
+        /// <summary>
+        /// If this is a member semantic model for executable code, then returns the syntax tree semantic model that was used to create this model.
+        /// If this is a syntax tree semantic model, then returns this instance.
+        /// Otherwise, returns null.
```

Comment:
>Otherwise, returns null. [](start = 12, length = 24)

This might not be  the right behavior for a speculative semantic model. I think it is not a syntax tree semantic model, so should probably return itself. Should add tests for scenarios like that.

Reply:
Given that, the name probably shouldn't imply that the result is a syntax tree semantic model 

---
In reply to: [196867601](https://github.com/dotnet/roslyn/pull/27922#discussion_r196867601) [](ancestors = 196867601)

Reply:
Speculative semantic model is also constructed off an SyntaxTreeSemanticModel, for example http://source.roslyn.io/#Microsoft.CodeAnalysis.CSharp/Compilation/AttributeSemanticModel.cs,36. For speculative semantic model, this property will return identical value as the public API [SemanticModel.ParentModel](http://source.roslyn.io/#Microsoft.CodeAnalysis/Compilation/SemanticModel.cs,266).

Reply:
Ah, I guess you meant if user invoked `GetOperation` on a speculative model? I will adjust accordingly and add tests.